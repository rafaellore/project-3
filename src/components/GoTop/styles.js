import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 2.5rem;
    height: 2.5rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    border-radius: 50%;
  `}
`;
