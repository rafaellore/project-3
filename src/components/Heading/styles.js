import styled, { css } from 'styled-components';

export const Title = styled.h1`
  ${({ theme, isLightText }) => css`
    color: ${isLightText ? theme.colors.white : theme.colors.primaryColor};
  `}
`;
