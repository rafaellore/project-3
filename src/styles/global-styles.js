// global styles is like a css reset
// we can use theme here too

import { createGlobalStyle } from 'styled-components';

export const GlobalStyles = createGlobalStyle`

  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  body {
    font-size: 1.6rem;
    font-family: 'Open Sans', sans-serif;
  }
  h1, h2, h3, h4, h5, h6 {
    font-family: 'Montserrat', sans-serif;
  }
`;
