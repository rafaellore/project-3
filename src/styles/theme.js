// here we set some constants related to styles, like fonts, colors etc
// this const is used on ThemeProvider of styled-components with allows you to use in the childrens styles

export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
    mediumGray: '#DDDDDD',
  },

  spacings: {
    xsmall: '0.5rem',
    small: '1rem',
    medium: '1.5rem',
    large: '2rem',
    xlarge: '2.5rem',
    xxlarge: '3rem',
    huge: '3.5rem',
    xhuge: '4rem',
  },

  font: {
    sizes: {
      xsmall: '0.5rem',
      small: '1rem',
      medium: '1.5rem',
      large: '2rem',
      xlarge: '2.5rem',
      xxlarge: '3rem',
      huge: '3.5rem',
      xhuge: '4rem',
    },
  },

  media: {
    lteMedium: '(max-width: 768px)',
  },
};
