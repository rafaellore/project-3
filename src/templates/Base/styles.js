import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    padding-top: 54px;

    @media ${theme.media.lteMedium} {
      padding-top: 0;
    }
  `}
`;
