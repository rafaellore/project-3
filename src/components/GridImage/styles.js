import styled, { css } from 'styled-components';

import { Container as Text } from '../Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${() => css`
    width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
