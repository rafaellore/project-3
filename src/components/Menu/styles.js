import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Heading/styles';

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    width: 100%;
    border-bottom: ${theme.colors.mediumGray};
    background: ${theme.colors.white};
    transition: all 300ms ease-in-out;

    // to import styles from another place and edit himself
    > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    // to import styles from another place and edit himself
    & ${Heading} {
      margin-top: 0;
      margin-bottom: 0;
    }

    //media querie
    @media ${theme.media.lteMedium} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;
      ${visible && menuVisible()}

      //styles from another place inside media querie
      > ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
      }

      & ${Heading} {
        padding-bottom: ${theme.spacings.large};
        margin: 0 auto;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    //media querie
    @media ${theme.media.lteMedium} {
      display: block;
      text-align: center;
      padding: ${theme.spacings.xxlarge} 0;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme, visible }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 2.5rem;
    height: 2.5rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    display: none;
    opacity: 0;
    visibility: hidden;

    // needs this to work
    pointer-events: ${visible ? 'none' : 'all'};

    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      justify-content: center;

      opacity: initial;
      visibility: visible;
    }

    > svg {
      width: 1.5rem;
      height: 1.5rem;
    }
  `}
`;
