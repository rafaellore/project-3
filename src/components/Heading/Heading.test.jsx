import { screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Heading } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<Heading/>', () => {
  it('should render with default values', () => {
    renderTheme(<Heading>Simple text</Heading>);

    const heading = screen.getByRole('heading', { name: 'Simple text' });

    expect(heading).toHaveStyle({
      color: theme.colors.primaryColor,
      'font-size': theme.font.sizes.xhuge,
      'text-transform': 'none',
    });
  });

  it('should render with correct with white text', () => {
    renderTheme(<Heading colorDark={false}>Simple text</Heading>);

    const heading = screen.getByRole('heading', { name: 'Simple text' });

    expect(heading).toHaveStyle({
      color: theme.colors.white,
      'font-size': theme.font.sizes.xhuge,
      'text-transform': 'none',
    });
  });

  it('should render correct heading sizes', () => {
    const { rerender } = renderTheme(
      <Heading size="small">Simple text</Heading>,
    );
    const heading = screen.getByRole('heading', { name: 'Simple text' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.medium,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="big">Simple text</Heading>
      </ThemeProvider>,
    );

    expect(screen.getByRole('heading', { name: 'Simple text' })).toHaveStyle({
      'font-size': theme.font.sizes.xlarge,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="medium">Simple text</Heading>
      </ThemeProvider>,
    );

    expect(screen.getByRole('heading', { name: 'Simple text' })).toHaveStyle({
      'font-size': theme.font.sizes.large,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="huge">Simple text</Heading>
      </ThemeProvider>,
    );

    expect(screen.getByRole('heading', { name: 'Simple text' })).toHaveStyle({
      'font-size': theme.font.sizes.xhuge,
    });
  });

  it('should render correct font-size when using mobile', () => {
    renderTheme(<Heading size="huge">Simple text</Heading>);

    const heading = screen.getByRole('heading', { name: 'Simple text' });

    // when we use media with toHaveStyleRule, is like to set an window size
    // and the another fields is the expects
    expect(heading).toHaveStyleRule('font-size', theme.font.sizes.xlarge, {
      media: theme.media.lteMedium,
    });
  });

  it('should render with uppercase letters', () => {
    renderTheme(
      <Heading size="huge" uppercase>
        Simple text
      </Heading>,
    );

    const heading = screen.getByRole('heading', { name: 'Simple text' });

    expect(heading).toHaveStyle({
      'text-transform': 'uppercase',
    });
  });

  it('should render correct heading element', () => {
    const { container } = renderTheme(<Heading as="h6">Simple text</Heading>);

    const h6 = container.querySelector('h6');

    expect(h6.tagName.toLowerCase()).toBe('h6');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Heading>Simple text</Heading>);

    // with firstChild this gets only the p tag
    expect(container).toMatchSnapshot();
  });
});
