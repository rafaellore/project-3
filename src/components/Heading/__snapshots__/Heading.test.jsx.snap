// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading/> should match snapshot 1`] = `
.c0 {
  color: #0A1128;
  font-size: 4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c0 {
    font-size: 2.5rem;
  }
}

<div>
  <h1
    class="c0"
  >
    Simple text
  </h1>
</div>
`;
